client/my-sites/site-settings/speed-up-site-settings.jsx

```diff
@@ -14,23 +14,63 @@ import { connect } from 'react-redux';
 import Card from 'components/card';
 import JetpackModuleToggle from 'my-sites/site-settings/jetpack-module-toggle';
 import FormFieldset from 'components/forms/form-fieldset';
+import CompactFormToggle from 'components/forms/form-toggle/compact';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
 import isJetpackModuleUnavailableInDevelopmentMode from 'state/selectors/is-jetpack-module-unavailable-in-development-mode';
 import isJetpackSiteInDevelopmentMode from 'state/selectors/is-jetpack-site-in-development-mode';
+import { activateModule, deactivateModule } from 'state/jetpack/modules/actions';
+import isJetpackModuleActive from 'state/selectors/is-jetpack-module-active';
+import isActivatingJetpackModule from 'state/selectors/is-activating-jetpack-module';
+import isDeactivatingJetpackModule from 'state/selectors/is-deactivating-jetpack-module';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { getSiteSlug } from 'state/sites/selectors';
 import SupportInfo from 'components/support-info';
 
 class SpeedUpSiteSettings extends Component {
+	static defaultProps = {
+		togglingSiteAccelerator: false,
+	};
+
 	static propTypes = {
 		fields: PropTypes.object,
 		isRequestingSettings: PropTypes.bool,
 		isSavingSettings: PropTypes.bool,
 		jetpackVersionSupportsLazyImages: PropTypes.bool,
+		siteAcceleratorSupported: PropTypes.bool,
+		activateModule: PropTypes.func,
+		deactivateModule: PropTypes.func,
 
 		// Connected props
 		photonModuleUnavailable: PropTypes.bool,
 		selectedSiteId: PropTypes.number,
 		siteSlug: PropTypes.string,
+		photonModuleActive: PropTypes.bool,
+		assetCdnModuleActive: PropTypes.bool,
+		togglingSiteAccelerator: PropTypes.bool,
+	};
+
+	handleCdnChange = () => {
+		const { photonModuleActive, assetCdnModuleActive, selectedSiteId } = this.props;
+
+		// Check if any of the CDN options are on.
+		const cdnStatus = photonModuleActive || assetCdnModuleActive;
+
+		// If one of them is on, we turn everything off.
+		if ( true === cdnStatus ) {
+			if ( false === ! photonModuleActive ) {
+				this.props.deactivateModule( selectedSiteId, 'photon' );
+			}
+			if ( false === ! assetCdnModuleActive ) {
+				this.props.deactivateModule( selectedSiteId, 'photon-cdn' );
+			}
+		} else {
+			if ( false === photonModuleActive ) {
+				this.props.activateModule( selectedSiteId, 'photon' );
+			}
+			if ( false === assetCdnModuleActive ) {
+				this.props.activateModule( selectedSiteId, 'photon-cdn' );
+			}
```

Comment:
I keep getting two success notices when toggling this:

![](https://cldup.com/eMkLAA1i85.png)

Perhaps we can resolve this by using the third (`silent`) argument of the `activateModule()` / `deactivateModule()` action creators?

Reply:
How should I go about this? I can't silent all of them otherwise there won't be any notice at all?

Reply:
This seems like something that shouldn't be handled in Calypso. We could always make the last one non-silent, but this won't necessarily work. 

Since these are formed as separate requests, what will happen if one of them fails? I keep thinking this should be handled at the Jetpack side. Jetpack should know what to activate and what to deactivate, without having to ask Calypso to do a bunch of remote requests and do the logic heavylifting, considering that it might even not have the latest activation state.

Reply:
Right; it almost seems like we would need to create a way to bulk activate / deactivate modules in one request.

Reply:
The settings endpoint (where modules are mapped to booleans in a flat object) allows multiple modules to be activated at once

Reply:
Quick update: I started working on a new action to bulk-activate modules here:
https://github.com/Automattic/wp-calypso/compare/add/site-accelerator-ui...add/bulk-module-management?expand=1

I'm running into a few issues so far, but @oskosk is helping me with it as well!

Reply:
Quick update: I started working on a new action to bulk-activate modules here:
https://github.com/Automattic/wp-calypso/compare/add/site-accelerator-ui...add/bulk-module-management?expand=1

I'm running into a few issues so far, but @oskosk is helping me with it as well!

Reply:
What about the settings endpoint as @oskosk suggests? It allows multiple settings to be updated and modules to be (de)activated at the same time with a single request. Is there a reason why we won't be using it for this?

Reply:
@oskosk Could you tell me more about that one when you get the chance?

Reply:
I have added this in 5c857672c5541f85bd4295cb0b5e9e7f869ddf21. This seems to work, but I get no notices at all now when toggling the main toggle.

Reply:
I tried 78c530cab6e92f444fe0b9ef344f8c37efb56b33 as a new way to maybe trigger notices when toggling the main toggle, but that does not seem to trigger any notices.

@tyxla As we discussed, I would appreciate your help with this tomorrow morning if you have a bit of time, to help me understand what I am doing wrong here.

Thank you!

Reply:
So, `setFieldValue()` has a third argument, `autosave`, which is `false` by default. When `true`, it will automatically save all the changed but unsaved settings.

Anyway, since this sets a single value, I'm afraid it might not work well for you if you want to set multiple values and save after that...

So you might want to go a level lower and use `wrapSettingsForm`'s `updateFields()` like this:

```
this.props.updateFields( {
	'module-1': true,
	'module-2': false,
}, () => {
	this.submitForm();
} );
```

This way you won't need to `saveJetpackSettings`, and notice will be taken care of by the `wrapSettingsForm` itself.

Let me know if that works!

Reply:
It seems to work well, thanks a lot! I added it in ed7ccddf89218f8c03dcf9d1e8543a7db1a45228 and everything appears to be working properly for me.